query/src/sessions/session_mgr.rs

```diff
@@ -69,6 +71,14 @@ impl SessionManager {
         // Cluster discovery.
         let discovery = ClusterDiscovery::create_global(conf.clone()).await?;
 
+        let storage_runtime = {
+            let mut storage_num_cpus = conf.storage.storage_num_cpus as usize;
+            if storage_num_cpus == 0 {
+                storage_num_cpus = std::cmp::max(1, num_cpus::get() / 2)
```

Comment:
why it defaults to half of the CPU, I think it might be higher than worker threads num?

Reply:
> I think it might be higher than worker threads num?

Not sure.

if underlying io tasks are executed in a blocking way, or some CPU-intensive tasks, let say, decompression, deserialization is also executed in these threads, it might be better to spawn more threads (when the physical resource is available).

In our case, IO operations are non-block, typically io threads just issue connect/read/write/etc, and wait them to be completed. Spawning lots of threads may not help.

Even worse, If the total num of io and computing threads is too larger(than the num of cores),  io and computing tasks are more likely to be scheduled in the same physical core -- which is this PR trying  to avoid.

> why it defaults to half of the CPU

This is an arbitrary number :D, during the benchmark(settings.max_threads = num_cpu), it seems to be a good/balanced num for the cases (but not the best for all the cases).  it would be better if this num could be chosen adaptively though。